File: Client/src/types/adminNotification.ts
export interface IAdminNotification {
  id: number;
  action: "ProductCreated" | "ProductDeleted" | string;
  productId: number;
  productName: string;
  vendorId?: string;
  vendorName?: string;
  message: string;
  createdAt: string;
  isRead: boolean;
}

File: API/RestoreAPI.Domain/Entities/AdminNotification.cs
namespace RestoreAPI.Domain.Entities;

public class AdminNotification
{
    public int Id { get; set; }
    public string Action { get; set; } = string.Empty; // "ProductCreated" or "ProductDeleted"
    public int ProductId { get; set; }
    public string ProductName { get; set; } = string.Empty;
    public string? VendorId { get; set; }
    public string? VendorName { get; set; }
    public string Message { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool IsRead { get; set; } = false;
}
